pandas/tests/indexing/multiindex/test_multiindex.py

```diff
@@ -98,3 +98,13 @@ def test_multiindex_with_datatime_level_preserves_freq(self):
         result = df.loc[0].index
         tm.assert_index_equal(result, dti)
         assert result.freq == dti.freq
+
+    def test_rename_multiindex_with_duplicates(self):
+        # GH 38015
+        idx = pd.Index([("A", "cat"), ("B", "cat"), ("B", "cat")])
```

Comment:
Looks good in General. Please use the MultiIndex constructor and compare the whole dataframe